app/dcs/api/s3/s3-compatibility: update with the latest storj.io/gateway/docs/s3-compatibility.md (#286)
storj/docs: app/dcs/api/s3/s3-compatibility/page.md

| HeadObject                                  | Full    |                                                              |                                                     |
| ListBucketAnalyticsConfigurations           | No      | No                                                           |                                                     |
| ListBucketIntelligentTieringConfigurations  | No      | No                                                           |                                                     |
| ListBucketInventoryConfigurations           | No      | No                                                           |                                                     |
| ListBucketMetricsConfigurations             | No      | No                                                           |                                                     |
| ListBuckets                                 | Full    |                                                              |                                                     |

> | ListMultipartUploads                        | Partial | Planned full. See https://github.com/storj/roadmap/issues/20 | See ListMultipartUploads section                    |

Reviewer: ```suggestion
| ListMultipartUploads                        | Partial | We may support lexicographical ordering in the future. | See ListMultipartUploads section                    |
```

| ListObjectVersions                          | No      | Planned. See https://github.com/storj/roadmap/issues/23      |                                                     |

> | ListObjects                                 | Partial | Planned full. See https://github.com/storj/roadmap/issues/20 | See ListObjects section                             |

Reviewer: ```suggestion
| ListObjects                                 | Partial | We may support lexicographical ordering in the future.  | See ListObjects section                             |
```

Reviewer: I think we should get rid of references to https://github.com/storj/roadmap/issues/20

> | ListObjectsV2                               | Partial | Planned full. See https://github.com/storj/roadmap/issues/20 | See ListObjects section                             |

Reviewer: ```suggestion
| ListObjectsV2                               | Partial | We may support lexicographical ordering in the future. | See ListObjects section                             |
```

| PutObjectRetention                          | No      | No                                                           |                                                     |
| PutObjectTagging                            | Full    |                                                              | Tags can be modified outside of tagging endpoints   |
| PutPublicAccessBlock                        | No      | No                                                           |                                                     |
| RestoreObject                               | No      | No                                                           |                                                     |
| SelectObjectContent                         | No      | No                                                           |                                                     |
| UploadPart                                  | Full    |                                                              |                                                     |

> | UploadPartCopy                              | No      | Planned. See https://github.com/storj/roadmap/issues/40      |                                                     |

Reviewer: ```suggestion
| UploadPartCopy                              | No      | Partial Support Planned. See https://github.com/storj/roadmap/issues/76      |                                                     |
```

except for features that rely on other actions that we haven't fully
implemented.

### GetBucketLocation

This is currently supported in Gateway-MT only.

#### LocationConstraint

> Specifies the placement region where the bucket data is stored:

Reviewer: ```suggestion
Specifies the placement region where the object data for a bucket is stored:
```

Reviewer: Should we specific OBJECT data? Given that this is grabbed from the satellite, this is more accurate right?

|     Value     |                                                        Description                                                        |
|:-------------:|:-------------------------------------------------------------------------------------------------------------------------:|

> | `global`      | Stored on the entire globe connected to the public network                                                                |

Reviewer: ```suggestion
| `global`      | Stored on nodes connected to the global public network                                                                |
```

Reviewer: It read a little bit weird. I think this is better, but I'm not SUPER opinionated here.
